airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/S3ConsumerFactory.kt

```diff
@@ -140,6 +123,41 @@ class S3ConsumerFactory {
         }
     }
 
+    fun createAsync(
+        outputRecordCollector: Consumer<AirbyteMessage>,
+        storageOps: S3StorageOperations,
+        s3Config: S3DestinationConfig,
+        catalog: ConfiguredAirbyteCatalog
+    ): SerializedAirbyteMessageConsumer? {
+        val writeConfigs = createWriteConfigs(storageOps, s3Config, catalog)
+        val createFunction =
+            getCreateFunction(
+                s3Config,
+                Function<String, BufferStorage> { fileExtension: String ->
+                    FileBuffer(fileExtension)
+                },
+                DEFAULT_NAMESPACE
+            )
+        return AsyncStreamConsumer(
+            outputRecordCollector,
+            onStartFunction(storageOps, writeConfigs),
+            onCloseFunction(storageOps, writeConfigs),
+            S3DestinationFlushFunction(
+                optimalBatchSizeBytes = FileBuffer.MAX_PER_STREAM_BUFFER_SIZE_BYTES,
+                {
+                    SerializedBufferingStrategy(
+                        createFunction,
+                        catalog,
+                        flushBufferFunction(storageOps, writeConfigs, catalog, DEFAULT_NAMESPACE)
+                    )
+                },
+                DEFAULT_NAMESPACE
+            ),
+            catalog,
+            BufferManager(DEFAULT_NAMESPACE)
```

Comment:
I don't know that we accept a defaultNamespace parameter in enough places yet...